Automattic/wp-calypso
# Blog onboarding: Offer option to logged users to create a new site

Related to https://github.com/Automattic/dotcom-forge/issues/3727

## Proposed Changes

Users currently can't use any of the Blog onboarding flows (start-writing / design-first) if they already have a site.
Now, we want to show a screen to select to use an existing not-launched site or create a new one.
As we can't check for existing sites and filter for not-launched, we created a new step that is only for users with sites.
If the user selects to use an existing site, we show all the not-launched sites, and then select one, we show a modal to confirm.
Copy in process, details: p1698771990960179/1698766219.275149-slack-CKZHG0QCR

### TO DO
- [x] Check CSS on desktop and mobile
- [x] Check for console error on site-selector step.
- [x] Remove subtitle on `new or existing` step.
- [ ] Unpublish posts when selecting an existing site.

### Screenshots
![image](https://github.com/Automattic/wp-calypso/assets/402286/aab9f3e7-374a-4d72-ba8e-d389bde94ba4)
![image](https://github.com/Automattic/wp-calypso/assets/402286/11b9e56c-0ba9-4287-a842-50b072a48ac8)

## Testing Instructions

* Go to `/setup/start-writing` with an account with 0 sites
* Go to `/setup/start-writing` with an account with > 0 sites not-launched
* Select to use an existing site
* Check the sites showing the site-selector step are correct
* If you have already launched sites, it should not show on this step
* Check all the process
* Same tests for `/setup/design-first`

## Pre-merge Checklist
